### DISCUSSION TITLE
Where to contribute?

### DISCUSSION THREAD

**Discussion by Arniiiii** [cppfastio/fast_io#775] [General]
Hello. I would like to update CMakeLists.txt . Would you explain in which branch should I try update and then make PR ?

**Comment by trcrsired** [comment]
next

Thank you

**Reply by Arniiiii** [reply]
I will need your help. There are too much missing include directives. Also most tests neither contain any kind of asserts nor  usage of a test framework. Could you help me with it soon?

**Reply by trcrsired** [reply]
missing include directives? what do you mean? ```#pragma once``` should be enough

**Reply by Arniiiii** [reply]
There's a lot of problems like "something is not defined" because of missing `#include "..."` in almost every header.

Could you help me ,please :
1. Where is [`output_stream` and `input_stream`](https://github.com/cppfastio/fast_io/blob/bbd98c96a34532ead365ade1e462242fd8af3c35/include/fast_io_core_impl/random_access_transmit.h#L9) defined? 
2. What is [this](https://github.com/cppfastio/fast_io/blob/next/include/fast_io_core_impl/literals/none_ascii.h)?

**Reply by trcrsired** [reply]
> There's a lot of problems like "something is not defined" because of missing `#include "..."` in almost every header.
> 
> Could you help me ,please :
> 
> 1. Where is [`output_stream` and `input_stream`](https://github.com/cppfastio/fast_io/blob/bbd98c96a34532ead365ade1e462242fd8af3c35/include/fast_io_core_impl/random_access_transmit.h#L9) defined?
> 2. What is [this](https://github.com/cppfastio/fast_io/blob/next/include/fast_io_core_impl/literals/none_ascii.h)?

Maybe you do not understand how ```#include``` works? It is literally just copying pasting. So they are not missing anything. The header you showed may need to be removed in the future.

2. That? It is for EBCDIC. Nothing wrong.
